roles/calico/README.rst

```diff
@@ -0,0 +1,147 @@
+about workloads, endpoints, and policy to each worker node. Endpoints are
+network interfaces associated with workloads. Calico is deployed in the Docker
+container on each worker node and managed by systemd. Any workload managed by
+Calico is registered as a service in Consul.
+
+Modes
+^^^^^
+
+Calico can run a public cloud environment that does not allow either L3 peering
+or L2 connectivity between Calico hosts. Calico will then route traffic between
+the Calico hosts using IP in IP mode. At this time, the full node-to-node BGP
+mesh is supported and configured in CiscoCloud only. Other cloud environments
+are set up with the IP in IP mode.
+
+Mesos
+^^^^^
+
+The systemd unit file for Mesos slave is updated with the new ``DOCKER_HOST``
+environment variable:
+
+.. code-block:: shell
+
+   DOCKER_HOST=localhost:2377
+
+This allows Calico to set up networking automatically by routing Docker API
+requests through the `Powerstrip <https://github.com/clusterhq/powerstrip>`_
+proxy that is running on port ``2377`` on each Mesos slave host. It sets
+a dependency, meaning if Calico Docker container is not running, the Mesos slave
+will refuse to start accordingly.
+
+Marathon
+^^^^^^^^
+
+When you start containers on top of Marathon, you will need to add two
+environment variables to your JSON file: ``CALICO_IP`` and ``CALICO_PROFILE``.
+You can assign an IP address to ``CALICO_IP`` explicitly or set ``auto`` and it
+will be allocated automatically. If the profile set with ``CALICO_PROFILE`` doesn't
+exist, it will be created automatically. If you don't provide the two variables,
+the Docker default network settings will be applied.
+
+Example:
+
+.. code-block:: json
+
+   {
+     "container": {
+       "type": "DOCKER",
+       "docker": {
+         "image": "busybox",
+         "parameters":[{"key": "env", "value": "CALICO_IP=auto"},
+                       {"key": "env", "value": "CALICO_PROFILE=dev"}]
+       }
+     },
+     "id": "testapp",
+     "instances": 1,
+     "cpus": 0.1,
+     "mem": 32,
+     "uris": [],
+     "cmd": "while sleep 10; do date -u +%T; done"
+   }
+
+Consul
+^^^^^^
+
+When you start a workload on Marathon with the proper environment variables
+as ``CALICO_IP`` and ``CALICO_PROFILE``, the workload is registered in Consul
+as a service. The Powerstrip logic was extended in this case.
+The registered name is constructed in this way: ``MARATHON_APP_ID`` plus
+``-calico`` suffix. For example, if you create a workload with the name of
+``testapp``, then the ``testapp-calico`` service will be registered in Consul.
+
+Thus, you have the option to query Consul in two ways:
+1. In order to obtain Docker host IP addresses where your workload is running:
+
+.. code-block:: shell
+
+   dig @localhost -p 8600 testapp.service.consul
+
+2. To resolve IP addresses from the Calico network:
+
+.. code-block:: shell
+
+   dig @localhost -p 8600 testapp-calico.service.consul
```

Comment:
I believe it would be possible to register all the services as "calico" with a tag of their actual name to get `testapp.calico.service.consul`... but then `calico.service.consul` would get all the IPs, which is maybe not desirable.

At any rate after more though, shouldn't we choose a name that's not tied to calico-the-implementation? Something like `testapp-direct` might be more readily understood if one doesn't know the technologies involved, and swappable for another component as well, if Calico doesn't work out.